src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -277,25 +278,66 @@ def __init__(self, **kwargs):
+                if 'compression_opts' in self.__iosettings:
+                    warnings.warn('Compression disabled by compression=False setting. ' +
+                                  'compression_opts parameter will, therefore, be ignored.')
+                    self.__iosettings.pop('compression_opts', None)
+        # Validate the compression options used
+        self._check_compression_options()
+        # Confirm that the compressor is supported by h5py
+        if not self.filter_available(self.__iosettings.get('compression', None)):
+            raise ValueError("%s compression not support by this version of h5py." %
+                             str(self.__iosettings['compression']))
+        # Check possible parameter collisions
+        if isinstance(self.data, Dataset):
+            for k in self.__iosettings.keys():
+                warnings.warn("%s in H5DataIO will be ignored with H5DataIO.data being an HDF5 dataset" % k)
+
+    def _check_compression_options(self):
+        """
+        Internal helper function used to check if compression options are compliant
+        with the compression filter used.
+
+        :raises ValueError: If incompatible options are detected
+        """
         if 'compression' in self.__iosettings:
-            if isinstance(self.__iosettings['compression'], bool):
-                if self.__iosettings['compression']:
-                    self.__iosettings['compression'] = 'gzip'
-                else:
-                    self.__iosettings.pop('compression', None)
-                    if 'compression_opts' in self.__iosettings:
-                        warnings.warn('Compression disabled by compression=False setting. ' +
-                                      'compression_opts parameter will, therefore, be ignored.')
-                        self.__iosettings.pop('compression_opts', None)
-        if 'compression' in self.__iosettings:
+            if 'compression_opts' in self.__iosettings:
+                if self.__iosettings['compression'] == 'gzip':
+                    if self.__iosettings['compression_opts'] not in range(10):
+                        raise ValueError("GZIP compression_opts setting must be an integer from 0-9, " +
+                                         "not " + str(self.__iosettings['compression_opts']))
+                elif self.__iosettings['compression'] == 'lzf':
+                    if self.__iosettings['compression_opts'] is not None:
+                        raise ValueError("LZF compression filter accepts no compression_opts")
+                elif self.__iosettings['compression'] == 'szip':
+                    if not isinstance(self.__iosettings['compression_opts'], tuple) or \
+                                    len(self.__iosettings['compression_opts']) != 2:
+                        raise ValueError("SZIP compression filter compression_opts" +
+                                         " must be a 2-tuple ('ec'|'nn', even integer 0-32")
```

Comment:
Super minor, but regarding code style, I'm pretty sure the backslash and plus symbols are not necessary. 

PEP8 says: "The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation."

Reply:
You need an ending parenthesis inside the last quote marker on line 322

Reply:
Since we check for gzip compression opts being an int from 0-9, can we also check for the szip compression opts being ('ec'|'nn', even integer 0-32)?

Reply:
Not sure what the issue with 322 is?

> szip compression opts being ('ec'|'nn', even integer 0-32)?

sure

Reply:
Not a syntax error - but within the string, the "(" before "'ec'" needs a corresponding ")".